src/hud_area_label.rs

```diff
@@ -0,0 +1,240 @@
+                if self.opening {
+                    self.frame_index += 1;
+                } else {
+                    self.frame_index -= 1;
+                }
+
+                Some(self)
+            }
+        } else {
+            Some(self)
+        }
+    }
+}
+
+/// Displays a temporary area label on top of the screen
+#[derive(Default)]
+pub struct HudAreaLabel {
+    new_label: Option<String>,
+    animation: Option<Animation>,
+    hide_timer: Option<Timer>,
+}
+
+impl HudAreaLabel {
+    fn is_fully_visible(&self) -> bool {
+        self.animation.is_some()
+    }
+
+    /// Schedules a new area title to be displayed
+    pub fn show_area_title<S: Into<String>>(&mut self, label: S) {
+        self.new_label = Some(label.into().to_uppercase());
+        self.animation = Some(Animation::new_opening());
+        self.hide_timer = Some(Timer::from_seconds(2.5, false));
+    }
+}
+
+/// Used to identify UI nodes belonging to the HUD area label
+pub struct HudAreaLabelNode;
+
+/// Updates the HUD area label system
+pub fn update_hud_area_label(
+    time: Res<Time>,
+    assets: Res<HudAreaLabelAssets>,
+    mut hud_label: ResMut<HudAreaLabel>,
+    mut text_query: Query<&mut Text, With<HudAreaLabelNode>>,
+    mut images: Query<&mut Handle<ColorMaterial>, With<HudAreaLabelNode>>,
+) {
+    let delta_time = time.delta_seconds();
+
+    if let Some(new_text) = hud_label.new_label.take() {
+        if let Some(mut text) = text_query.iter_mut().next() {
+            text.sections[0].value = new_text;
+        }
+    }
+
+    if let Some(animation) = hud_label.animation.take() {
+        hud_label.animation = animation.tick(delta_time);
+
+        if let Some(animation) = &hud_label.animation {
+            for mut image in images.iter_mut() {
+                *image = assets.hud_area_border_frames[animation.frame_index].clone();
+            }
+
+            for mut text in text_query.iter_mut() {
+                let opacity = animation.frame_index as f32 / BORDER_ANIMATION_FRAMES as f32;
+                text.sections[0].style.color = Color::rgba(1.0, 1.0, 1.0, opacity);
+            }
+        } else if hud_label.is_fully_visible() {
+            for mut image in images.iter_mut() {
+                *image = assets.hud_area_border_frames[BORDER_ANIMATION_FRAMES - 1].clone();
+            }
+        }
+    }
+
+    if let Some(hide_timer) = &mut hud_label.hide_timer {
+        hide_timer.tick(delta_time);
+
+        if hide_timer.just_finished() {
+            hud_label.animation = Some(Animation::new_closing());
+        }
+    }
+}
+
+/// Sets up the components necessary for the HUD area label
+pub fn setup_hud_area_label(commands: &mut Commands, assets: Res<HudAreaLabelAssets>) {
+    commands
+        // Root container at the top of the screen
+        .spawn(NodeBundle {
+            style: Style {
+                position_type: PositionType::Absolute,
+                position: Rect {
+                    top: Val::Px(40.0),
```

Comment:
Can these magic numbers should be factored out?

Reply:
All(?) magic numbers are now documented constants at the top of the file.